dask/array/fft.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import absolute_import
+
+import numpy as np
+import numpy.fft as npfft
+
+from .core import map_blocks
+
+"""
+TODO: implement `n` kwarg of fft's. With the current approach this would mean
+specifying a how the chunks change to pass to map_blocks.
+"""
+
+
+def fft(a, axis=-1):
+    """
+    Compute the one-dimensional discrete Fourier Transform along an axis that
+    only has one chunk.
+
+    Parameters
+    ----------
+    a : dask.array
+        input array, can be complex
+    axis : int, optional
+        Axis over which to compute the FFT.  If not given, the last
+        axis is used.
+    """
+    if len(a.chunks[axis]) != 1:
+        raise ValueError("Dask array only supports indexing along an axis"
+                         "that has a single chunk. You tried FFTing axis %s"
+                         " which has chunks %s." % (axis, a.chunks[axis]))
```

Comment:
We should point users to `rechunk` or ask them to specify `chunks` differently when they create their array, perhaps giving a small example.  The phrase "indexing along an axis" isn't immediately clear to me.  I would stay away from using FFT as a verb.  Crafting this error message is probably one of the more important parts of this PR.  Given that it will be used fairly often we might want to construct it separately.